api/config_entry_gateways.go

```diff
@@ -67,19 +67,25 @@ type IngressService struct {
 	ServiceSubset string
 }
 
-func (i *IngressGatewayConfigEntry) GetKind() string {
+func (i IngressService) DecodeKeyMapping() map[string]string {
+	return map[string]string{
+		"service_subset": "servicesubset",
+	}
+}
+
+func (i IngressGatewayConfigEntry) GetKind() string {
```

Comment:
Shouldn't these have the same signature as all the other top-level config entry types such as the ServiceResolverConfigEntry

Reply:
Do you mean the method receiver is not a pointer?

I think all of these should be using value receivers. They do not modify the struct, they are data records, not objects with state. A pointer to this type still implements the same interface.